app/objects/beatmap.py

```diff
@@ -737,6 +740,18 @@ async def _update_if_available(self) -> None:
                 {"set_id": self.id},
             )
 
+    async def force_update(self) -> None:
+        await self._update_if_available()
```

Comment:
you should check that the map did actually change before deleting cache and/or redownloading the .osu file(s). you should also repopulate the cache in the case it did change

Reply:
to update a map forced seems unnecessary because we had the expiration system (2 hours + 5 hours per year).
but the automaticlly check and update map function seems not works in my instance.
i will check and find another solution if i find something weird.